sw/device/silicon_creator/mask_rom/mask_rom_epmp.S

```diff
@@ -117,7 +120,10 @@ mask_rom_epmp_init:
            CFG_INDEX(2  % 4, EPMP_CFG_A_NAPOT | EPMP_CFG_LR)    /* ROM         */
   li   t1, CFG_INDEX(5  % 4, EPMP_CFG_A_NAPOT | EPMP_CFG_LR)    /* eFLASH      */
   li   t2, CFG_INDEX(11 % 4, EPMP_CFG_A_TOR   | EPMP_CFG_LRW)   /* MMIO        */
-  li   t3, CFG_INDEX(14 % 4, EPMP_CFG_A_NA4   | EPMP_CFG_L)   | /* Stack Guard */ \
+  li   t3, CFG_INDEX(13 % 4, EPMP_CFG_A_NAPOT | EPMP_CFG_R    |                   \
+                                                EPMP_CFG_W    |                   \
+                                                EPMP_CFG_X)   | /* Debug ROM   */ \
```

Comment:
Hmm, why doesn't this use `EPMP_CFG_LRWX`?

Reply:
This region seems to execute code and do rd/wr in the same area.  I didn't dig into the details of the debug rom area, but I'll see if I can dig into it.

Reply:
I wasn't able to find the source for the debug rom, but here is the trace when a breakpoint is hit:

`         568902      91227      0001035c        00100073        ebreak  
         568922      91237      00010800        00c0006f        jal     x0,1080c          x0=0x00000000
         568938      91245      0001080c        0ff0000f        fence   iorw,iorw       
         568942      91247      00010810        7b241073        csrrw   x0,dscratch,x8    x8:0x00004040  x0=0x00000000
         568946      91249      00010814        7b351073        csrrw   x0,0x7b3,x10     x10:0x10000748  x0=0x00000000
         568950      91251      00010818        00000517        auipc   x10,0x0  x10=0x00010818
         568954      91253      0001081c        00c55513        srli    x10,x10,0xc      x10:0x00010818 x10=0x00000010
         568958      91255      00010820        00c51513        slli    x10,x10,0xc      x10:0x00000010 x10=0x00010000
         568962      91257      00010824        f1402473        csrrs   x8,mhartid,x0     x0:0x00000000  x8=0x00000000
         568978      91265      00010828        10852023        sw      x8,256(x10)      x10:0x00010000  x8:0x00000000 PA:0x00010100 store:0x00000000 load:0x00000000
         568980      91266      0001082c        00a40433        add     x8,x8,x10         x8:0x00000000 x10:0x00010000  x8=0x00010000
         568994      91273      00010830        40044403        lbu     x8,1024(x8)       x8:0x00010000  x8=0x00000000 PA:0x00010400 store:0x00000000
         568998      91275      00010834        00147413        andi    x8,x8,1   x8:0x00000000  x8=0x00000000
         569000      91276      00010838        02041c63        bne     x8,x0,10870       x8:0x00000000  x0:0x00000000
         569002      91277      0001083c        f1402473        csrrs   x8,mhartid,x0     x0:0x00000000  x8=0x00000000
         569004      91278      00010840        00a40433        add     x8,x8,x10         x8:0x00000000 x10:0x00010000  x8=0x00010000
         569014      91283      00010844        40044403        lbu     x8,1024(x8)       x8:0x00010000  x8=0x00000000 PA:0x00010400 store:0x00000000
         569018      91285      00010848        00247413        andi    x8,x8,2   x8:0x00000000  x8=0x00000000
         569020      91286      0001084c        fa041ce3        bne     x8,x0,10804       x8:0x00000000  x0:0x00000000
         569022      91287      00010850        fd5ff06f        jal     x0,10824          x0=0x00000000
         569034      91293      00010824        f1402473        csrrs   x8,mhartid,x0     x0:0x00000000  x8=0x00000000
         569050      91301      00010828        10852023        sw      x8,256(x10)      x10:0x00010000  x8:0x00000000 PA:0x00010100 store:0x00000000 load:0x00000000
         569052      91302      0001082c        00a40433        add     x8,x8,x10         x8:0x00000000 x10:0x00010000  x8=0x00010000
         569066      91309      00010830        40044403        lbu     x8,1024(x8)       x8:0x00010000  x8=0x00000001 PA:0x00010400 store:0x00000000
         569070      91311      00010834        00147413        andi    x8,x8,1   x8:0x00000001  x8=0x00000001
         569072      91312      00010838        02041c63        bne     x8,x0,10870       x8:0x00000001  x0:0x00000000
         569098      91325      00010870        10052223        sw      x0,260(x10)      x10:0x00010000  x0:0x00000000 PA:0x00010104 store:0x00000000 load:0x00000000
         569100      91326      00010874        7b302573        csrrs   x10,0x7b3,x0      x0:0x00000000 x10=0x10000748
         569102      91327      00010878        7b202473        csrrs   x8,dscratch,x0    x0:0x00000000  x8=0x00004040
         569104      91328      0001087c        a85ff06f        jal     x0,10300          x0=0x00000000
         569118      91335      00010300        0380006f        jal     x0,10338          x0=0x00000000
         569134      91343      00010338        7b351073        csrrw   x0,0x7b3,x10     x10:0x10000748  x0=0x00000000
         569138      91345      0001033c        00000517        auipc   x10,0x0  x10=0x0001033c
         569142      91347      00010340        00c55513        srli    x10,x10,0xc      x10:0x0001033c x10=0x00000010
         569146      91349      00010344        00c51513        slli    x10,x10,0xc      x10:0x00000010 x10=0x00010000
         569162      91357      00010348        38052403        lw      x8,896(x10)      x10:0x00010000  x8=0x00000000 PA:0x00010380 store:0x00000000
         569164      91358      0001034c        00000013        addi    x0,x0,0   x0:0x00000000  x0=0x00000000
         569166      91359      00010350        00000013        addi    x0,x0,0   x0:0x00000000  x0=0x00000000
         569168      91360      00010354        00000013        addi    x0,x0,0   x0:0x00000000  x0=0x00000000
         569170      91361      00010358        7b302573        csrrs   x10,0x7b3,x0      x0:0x00000000 x10=0x10000748
`
You can see there are loads and stores to the same page.

Reply:
> I wasn't able to find the source for the debug rom
Have a look here: https://github.com/lowRISC/opentitan/tree/master/hw/vendor/pulp_riscv_dbg/debug_rom

Reply:
@imphil Thanks

Ok, looks like there are a few implementation specific registers int the dm_rom area that get read/written:

// These are implementation-specific addresses in the Debug Module
#define HALTED    0x100
#define GOING     0x104
#define RESUMING  0x108
#define EXCEPTION 0x10C

hence the need to add RW access in addition to execute.

Reply:
I didn't want to lock the entry so I could change it once I determined the life cycle.

Reply:
Oh I see, sorry, I missed that detail.

You can safely set the L bit here because the RLB (rule locking bypass) bit is set while the ROM code executes (see the ePMP spec [here](https://github.com/riscv/riscv-tee/blob/61455747230a26002d741f64879dd78cc9689323/Smepmp/Smepmp.pdf)). In fact I think you need the L bit to be set otherwise the rule won't be applied in M mode and the ROM only runs in M mode.

Reply:
Table 2.1 does indicate that region is inaccessible in M mode, but I am able to access DM_ROM with RWX bits set. My concern is the footnote to table 2.1:

*: Locked rules cannot be removed or modified until a PMP reset, unless mseccfg.RLB is set.

Is RLB set during boot of mask_rom?

Reply:
Yeah RLB is set for the duration of mask ROM execution (now in the reset values).

Reply:
Ok, I think I want to keep this configuration as set.